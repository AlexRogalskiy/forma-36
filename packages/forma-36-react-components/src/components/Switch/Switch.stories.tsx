import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Switch from './Switch';

storiesOf('Components|Switch', module)
  .addParameters({
    propTypes: Switch['__docgenInfo'],
  })
  .add('default', () => (
    <Switch
      className={text('className', '')}
      isChecked={boolean('isChecked', false)}
      isDisabled={boolean('isDisabled', false)}
      labelText={text('labelText', 'My label text')}
      onToggle={action('onToggle')}
    />
  ));
