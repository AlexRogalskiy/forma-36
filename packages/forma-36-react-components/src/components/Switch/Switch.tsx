import React, { FunctionComponent, KeyboardEvent } from 'react';
import FormLabel from '../Typography/Paragraph';
import classNames from 'classnames';
import styles from './Switch.css';

export interface SwitchProps {
  onToggle?: (isChecked: boolean) => void;
  isChecked?: boolean;
  isDisabled?: boolean;
  labelText: string;
  className?: string;
}

export const Switch: FunctionComponent<SwitchProps> = (props: SwitchProps) => {
  return (
    <div className={classNames(styles['Switch__wrapper'], props.className)}>
      <input
        type="checkbox"
        onClick={onToggle}
        onKeyUp={onKeyUp}
        checked={props.isChecked}
        disabled={props.isDisabled}
        className={classNames(styles['Switch'], {
          [styles['Switch--checked']]: props.isChecked,
          [styles['Switch--disabled']]: props.isDisabled,
        })}
      />
      <FormLabel
        className={classNames(styles['Switch__label'], {
          [styles['Switch__label--disabled']]: props.isDisabled,
        })}
      >
        {props.labelText}
      </FormLabel>
    </div>
  );

  function onToggle() {
    if (props.onToggle) {
      props.onToggle(!props.isChecked);
    }
  }

  function onKeyUp(e: KeyboardEvent<HTMLElement>) {
    if (props.onToggle && e.key === 'Enter') {
      props.onToggle(!props.isChecked);
    }
  }
};

export default Switch;
