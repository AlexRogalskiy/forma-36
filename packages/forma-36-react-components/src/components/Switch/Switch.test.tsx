import React from 'react';
import { shallow } from 'enzyme';
import Switch from './Switch';

it('renders the component', () => {
  const output = shallow(<Switch labelText="foobar" />);

  expect(output).toMatchSnapshot();
});

it('renders the component as checked', () => {
  const output = shallow(<Switch labelText="foobar" isChecked />);

  expect(output).toMatchSnapshot();
});

it('renders the component as disabled', () => {
  const output = shallow(<Switch labelText="foobar" isDisabled />);

  expect(output).toMatchSnapshot();
});

it('renders the component as disabled and checked', () => {
  const output = shallow(<Switch labelText="foobar" isDisabled isChecked />);

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <Switch labelText="foobar" className="my-extra-class" />,
  );

  expect(output).toMatchSnapshot();
});
