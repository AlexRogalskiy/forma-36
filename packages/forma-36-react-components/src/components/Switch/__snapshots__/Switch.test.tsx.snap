// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="Switch__wrapper"
>
  <input
    className="Switch"
    onClick={[Function]}
    onKeyUp={[Function]}
    type="checkbox"
  />
  <Paragraph
    className="Switch__label"
    element="p"
    testId="cf-ui-paragraph"
  >
    foobar
  </Paragraph>
</div>
`;

exports[`renders the component as checked 1`] = `
<div
  className="Switch__wrapper"
>
  <input
    checked={true}
    className="Switch Switch--checked"
    onClick={[Function]}
    onKeyUp={[Function]}
    type="checkbox"
  />
  <Paragraph
    className="Switch__label"
    element="p"
    testId="cf-ui-paragraph"
  >
    foobar
  </Paragraph>
</div>
`;

exports[`renders the component as disabled 1`] = `
<div
  className="Switch__wrapper"
>
  <input
    className="Switch Switch--disabled"
    disabled={true}
    onClick={[Function]}
    onKeyUp={[Function]}
    type="checkbox"
  />
  <Paragraph
    className="Switch__label Switch__label--disabled"
    element="p"
    testId="cf-ui-paragraph"
  >
    foobar
  </Paragraph>
</div>
`;

exports[`renders the component as disabled and checked 1`] = `
<div
  className="Switch__wrapper"
>
  <input
    checked={true}
    className="Switch Switch--checked Switch--disabled"
    disabled={true}
    onClick={[Function]}
    onKeyUp={[Function]}
    type="checkbox"
  />
  <Paragraph
    className="Switch__label Switch__label--disabled"
    element="p"
    testId="cf-ui-paragraph"
  >
    foobar
  </Paragraph>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
<div
  className="Switch__wrapper my-extra-class"
>
  <input
    className="Switch"
    onClick={[Function]}
    onKeyUp={[Function]}
    type="checkbox"
  />
  <Paragraph
    className="Switch__label"
    element="p"
    testId="cf-ui-paragraph"
  >
    foobar
  </Paragraph>
</div>
`;
